src/main/java/com/coherentsolutions/training/aqa/java/zhavrid/util/WebDriverSingleton.java

```diff
@@ -1,41 +1,63 @@
 package com.coherentsolutions.training.aqa.java.zhavrid.util;
 
 import com.coherentsolutions.training.aqa.java.zhavrid.constants.TimeConstants;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.MutableCapabilities;
 import org.openqa.selenium.PageLoadStrategy;
+import org.openqa.selenium.Platform;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
 import org.openqa.selenium.remote.RemoteWebDriver;
 
+import java.io.IOException;
 import java.net.URL;
 import java.time.Duration;
 import java.net.MalformedURLException;
 
 public class WebDriverSingleton {
-    public static WebDriver driver = null;
+    public static WebDriver driver;
     public static ChromeOptions options;
 
+    private static final String USERNAME_SL_KEY = "username";
```

Comment:
looks like local and remote driver setup becomes different. I propose to split this logic into 2 classes - LocalDriverFactory and RemoteDriverFactory